guava/src/com/google/common/collect/RegularImmutableSortedSet.java

```diff
@@ -162,9 +162,7 @@ public boolean equals(@Nullable Object object) {
     if (SortedIterables.hasSameComparator(comparator, that)) {
       Iterator<?> otherIterator = that.iterator();
       try {
-        Iterator<E> iterator = iterator();
-        while (iterator.hasNext()) {
-          Object element = iterator.next();
+        for (E element : this) {
```

Comment:
Exhaustive, definitive, guaranteed-correct analysis of this change by me and one other guy shows that the code was better how it was.

Or, you know, we just kind of prefer that way :-P We like the parallel `next()` calls. I'm submitting the change internally (minus the one part here), and we'll mirror it out shortly.